[FEAT] Dynamic Island Gameboard

Polysemouse opened the pull request:
# Description

Based on #2088

As the island is getting more crowded as more expansions are added, I hope the team can reconsider the PR.

- introduce dynamic island gameboard
  - every element outside of main island will be pushed outward as the land expands (except frogs, which keeps the current implementation for now)
  - gameboard expands as island expands.  No more running out of island space
- fix seal overlapping snorkeler and island elements
- move seal and frog popover to top of sea creature

Before|After
---|---
![image](https://github.com/sunflower-land/sunflower-land/assets/107602352/17ead383-57e8-4d7c-afff-9d3a042d2cdb)|![image](https://github.com/sunflower-land/sunflower-land/assets/107602352/3d050a3e-e8dc-4863-a987-1094b3c393be)

## Type of change

- [x] New feature (non-breaking change which adds functionality)

# How Has This Been Tested?

- look at layout of island at different expansion levels
- have 18 seals and some frogs in your island
- To test the farm with frogs and seals, temporary replace const data = await loadFrogs({ loadIncubatedFrogs: true }); in src\features\community\components\ProjectDignityFrogs.tsx with:
```tsx
      const data = Array.from({ length: 7 }, (_, i) => ({
        name: `Frog Name #${i + 1}`,
        pixel_image: `https://raw.githubusercontent.com/tiffanydys/frogs/main/pixel_images/${i + 1}.gif`,
        attributes: [],
        rarity: {
          weight: 1,
          name: `Rarity ${i + 1}`,
          visibleFrogs: 7,
          color: 'red'
        }
      }));
```
  - and replace const data = await loadSeals(); in src\features\community\components\ProjectDignitySeals.tsx with:
```tsx
      const data = Array.from({ length: 18 }, (_, i) => ({
        name: `Seal Name #${i + 1}`,
        pixel_image: `https://raw.githubusercontent.com/tiffanydys/seals/main/pixel_images/${i + 1}.png`,
        attributes: [],
        rarity: {
          weight: 1,
          name: `Rarity ${i + 1}`,
          visibleSeals: 18,
          color: 'red'
        }
      }));
```

# Checklist:

- [x] Title of the PR is relevant and is prefixed with [FEAT], [CHORE] or [FIX]
- [x] Screenshot if it includes any UI changes
- [x] I have read the contributing guidelines and agree to the T&Cs
- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] My changes generate no new warnings
- [x] New and existing unit tests pass locally with my changes

adamhannigan reviewed (commented):
Is it possible to bring these a bit closer? At 17 expansions the islands feel quite distant from the land.

For current players it may feel like they have disappeared.
<img width="916" alt="Screen Shot 2023-06-02 at 4 19 11 pm" src="https://github.com/sunflower-land/sunflower-land/assets/11745561/8f5b58be-ec9a-425b-82e4-59eaa7aa91fa">

Polysemouse commented:
@adamhannigan I moved islands closer to each other and updated the screenshot.  I checked every expansion level and islands are not overlapping with the main island
